### PR TITLE
Add Arm Neon optimizations

### PR DESCRIPTION
**Pull request by chalice19**
### Changelog category:
- Performance Improvement

### Changelog entry:

Add Arm Neon optimizations to some files where SSE optimization is implemented.

Optimization in src/Functions/LowerUpperImpl.h doesn't show good results yet.

### CODE REVIEW

File: cmake/cpu_features.cmake [hunk 1]
```diff
@@ -65,7 +65,7 @@ else ()
     endif ()
 
     set (TEST_FLAG "-msse4.2")
-    set (CMAKE_REQUIRED_FLAGS "${TEST_FLAG} -O0")
```
Comment (alexey-milovidov) [inline cmake/cpu_features.cmake:68]: This is wrong.

File: src/Columns/ColumnsCommon.h [hunk 2]
```diff
@@ -96,13 +113,13 @@ ColumnPtr selectIndexImpl(const Column & column, const IColumn & indexes, size_t
         throw Exception(ErrorCodes::SIZES_OF_COLUMNS_DOESNT_MATCH,
             "Size of indexes ({}) is less than required ({})", indexes.size(), limit);
 
-    if (const auto * data_uint8 = detail::getIndexesData<UInt8>(indexes))
+    if (auto * data_uint8 = detail::getIndexesData<UInt8>(indexes))
```
Comment (alexey-milovidov) [inline src/Columns/ColumnsCommon.h:116]: ?

**Comment by alexey-milovidov** [comment]
4% performance improvement, up to 1.5 times in some queries:

Screenshot_20220527_000741

**Comment by qoega** [comment]
If you will update this branch, there will be new performance test with aarch64. Just added recently.

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Comment by alexey-milovidov** [comment]
Merged in #38093.

**Pull request closed without merging**